product_multi_ean/models/product_product.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+# © 2012-2014 Guewen Baconnier (Camptocamp SA)
+# © 2015 Roberto Lizana (Trey)
+# © 2016 Pedro M. Baeza
+# © 2017 Dionisius M. (Portcities)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import Warning as UserError
+
+
+class ProductEan13(models.Model):
+    _name = 'product.ean13'
+    _description = "List of EAN13 for a product."
+    _order = 'sequence, id'
+
+    name = fields.Char(string='EAN13', size=13, required=True)
+    sequence = fields.Integer(string='Sequence', default=0)
+    product_id = fields.Many2one(
+        string='Product', comodel_name='product.product', required=True)
+
+    @api.multi
+    @api.constrains('name')
+    @api.onchange('name')
+    def _check_name(self):
+        for record in self:
+            if record.name and not self.env['barcode.nomenclature'].check_ean(
+                    record.name):
+                raise UserError(
+                    _('You provided an invalid "EAN13 Barcode" reference. You '
+                      'may use the "Internal Reference" field instead.'))
+
+    @api.multi
+    @api.constrains('name')
+    def _check_duplicates(self):
+        for record in self:
+            eans = self.search(
+                [('id', '!=', record.id), ('name', '=', record.name)])
+            if eans:
+                raise UserError(
+                    _('The EAN13 Barcode "%s" already exists for product '
+                      '"%s"') % (record.name, eans[0].product_id.name))
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    ean13_ids = fields.One2many(
+        comodel_name='product.ean13', inverse_name='product_id',
+        string='EAN13')
+    ean13 = fields.Char(
+        string='Main EAN13', compute='_compute_ean13', store=True,
+        inverse='_inverse_ean13', readonly=True)
+    barcode = fields.Char(
```

Comment:
AFAIK, barcode is now, not surely an ean13.
what about to create a model product.barcode and an one2many barcode_ids ?

Reply:
I'm not really sure about changing the model since the original module (or at least, the version for v8) only specify ean13. It may or may not be the original intent of the original developer.

Also, I never worked with PoS before so I'm not sure if this will work with PoS.